### PR TITLE
[GSOC-79] Sea pressure plugin

### PR DESCRIPTION
**Pull request by GauravKumar9920**
This PR introduces the following features - 

1.  Addition of Sea Pressure sensor plugin in ROS 2 jazzy and gazebo harmonic 
2. Updated SDF with added  Sea Pressure sensor Plugin.

Documentation on how to use the plugin can be found [here](https://yeongdocat.notion.site/SubSeaPressureSensor-53e0d4313a5c49698431421c97a56e9f?pvs=4).

### CODE REVIEW

File: models/dave_robot_models/description/rexrov/model.sdf [hunk 1]
```diff
@@ -584,5 +584,22 @@
       <joint_name>thruster8_joint</joint_name>
     </plugin>
 
+    <!-- Sea Pressure Sensor Plugin -->
+    <plugin
+      filename="sea_pressure_sensor"
+      name=" dave_gz_sensor_plugins::SubseaPressureSensorPlugin">
+      <namespace>rexrov</namespace>
```
Comment (woensug-choi) [inline models/dave_robot_models/description/rexrov/model.sdf:591]: Isn't this suppose to be make msg name as `/rexrov/sensor_output_topic`? Meanwhile, shouldn't `sensor_output_topic` be `pressure`? will we include other sensor msgs into this `sensor_output_topic`?

File: models/dave_robot_models/description/rexrov/model.sdf [hunk 2]
```diff
@@ -584,5 +584,22 @@
...
+      <!-- <link_name>base_link</link_name> -->
+      <topic>sea_pressure</topic>
+      <!-- <mass>0.015</mass> -->
```
Comment (woensug-choi) [inline models/dave_robot_models/description/rexrov/model.sdf]: Is this `mass` needed anytime later? if not should be removed

File: models/dave_robot_models/description/rexrov/model.sdf [hunk 3]
```diff
@@ -584,5 +584,22 @@
       <joint_name>thruster8_joint</joint_name>
     </plugin>
 
+    <!-- Sea Pressure Sensor Plugin -->
+    <plugin
+      filename="sea_pressure_sensor"
+      name=" dave_gz_sensor_plugins::SubseaPressureSensorPlugin">
+      <namespace>rexrov</namespace>
+      <!-- <link_name>base_link</link_name> -->
```
Comment (woensug-choi) [inline models/dave_robot_models/description/rexrov/model.sdf]: I this `link_name` needed anytime later? if not should be removed

File: models/dave_robot_models/description/rexrov/model.sdf [hunk 4]
```diff
@@ -584,5 +584,22 @@
...
+      <topic>sea_pressure</topic>
+      <!-- <mass>0.015</mass> -->
+      <update_rate>10</update_rate>
+      <!-- <saturation>3000</saturation> -->
+      <noise_sigma>3.0</noise_sigma>
+      <noise_amplitude>0.0</noise_amplitude>
+      <estimate_depth_on>false</estimate_depth_on>
```
Comment (woensug-choi) [inline models/dave_robot_models/description/rexrov/model.sdf]: I've treid `true` and nothing is added to the published msg. I suppose this was to add estimated depth value to the msg but it did not. please check.

File: gazebo/dave_gz_sensor_plugins/src/sea_pressure_sensor.cc [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+void SubseaPressureSensorPlugin::PreUpdate(
+  const gz::sim::UpdateInfo & _info, gz::sim::EntityComponentManager & _ecm)
+{
+  // Get model pose
+
+  gz::math::Pose3d sea_pressure_sensor_pos = GetModelPose(this->dataPtr->modelEntity, _ecm);
+
+  // const gz::sim::Entity & _entity = _ecm.EntityByComponents(gz::sim::components::WorldPose());
+  // gz::math::Vector3 pos;
+  // auto pos = GetWorldPose(_entity, _ecm);
```
Comment (woensug-choi) [inline gazebo/dave_gz_sensor_plugins/src/sea_pressure_sensor.cc]: Please remove any of the unnecessary codes. Many other exists in this file.

**Review by woensug-choi — CHANGES_REQUESTED** [review]
Almost there!

I would recommend using generic `FluidPressure` ROS msg (https://docs.ros2.org/latest/api/sensor_msgs/msg/FluidPressure.html) for pressure and for the estimated depth, make a new topic named `pressure_depth` similar fashion as https://wiki.ros.org/hector_quadrotor_gazebo_plugins which uses http://docs.ros.org/en/api/geometry_msgs/html/msg/PointStamped.html

File: models/dave_robot_models/description/rexrov/model.sdf (hunk 2 above)
Comment (GauravKumar9920) [reply models/dave_robot_models/description/rexrov/model.sdf]: Yess, we also have a pressure sensor model, I will add it in another PR in the future.
For now I have just added it as we might need it later, Shall I remove it for this PR and add it once the Model is Migrated ?

File: models/dave_robot_models/description/rexrov/model.sdf (hunk 3 above)
Comment (GauravKumar9920) [reply models/dave_robot_models/description/rexrov/model.sdf]: same as above

File: models/dave_robot_models/description/rexrov/model.sdf (hunk 4 above)
Comment (GauravKumar9920) [reply models/dave_robot_models/description/rexrov/model.sdf]: Fixed this issue in the upcoming commits : )

File: gazebo/dave_gz_sensor_plugins/src/sea_pressure_sensor.cc (hunk 5 above)
Comment (GauravKumar9920) [reply gazebo/dave_gz_sensor_plugins/src/sea_pressure_sensor.cc]: Done !!

File: models/dave_robot_models/description/rexrov/model.sdf (hunk 1 above)
Comment (GauravKumar9920) [reply models/dave_robot_models/description/rexrov/model.sdf:591]: Yes sir, you're correct.

**Review by woensug-choi — APPROVED** [review]

**Comment by woensug-choi** [comment]
@GauravKumar9920 I've confirmed all fixed and it runs fine :) You may click merge :D

**Pull request merged**